fork/fork.go

```diff
@@ -33,12 +33,12 @@ var (
 
 // Rules returns the hard fork information
 // CAUTIOUS: Use it when chainConfig value is not reachable
-func Rules(blockNumber *big.Int) (*params.Rules, error) {
+func Rules(blockNumber *big.Int) *params.Rules {
 	if hardForkBlockNumberConfig == nil {
-		return nil, errors.New("hardForkBlockNumberConfig variable is not initialized")
+		panic("fork.Rules should never be called before hardForkBlockNumberConfig initialization.")
 	}
 	rules := hardForkBlockNumberConfig.Rules(blockNumber)
-	return &rules, nil
+	return &rules
 }
 
 // SetHardForkBlockNumberConfig sets values in HardForkConfig if it is not nil.
```

Comment:
There is something wrong with the function. 
The config can be set only once, but it is called with different values : 
```
fork.SetHardForkBlockNumberConfig(&params.ChainConfig{IstanbulCompatibleBlock: istanbul.TestBlockHF})
```
```
fork.SetHardForkBlockNumberConfig(&params.ChainConfig{})
```
We don't know which one will be set.

Reply:
@winnie-byun It can be called at test.
```
// CAUTIOUS: Calling this function is can be dangerous, so avoid using it except tests
```